packages/castmill/lib/castmill/resources.ex

```diff
@@ -706,17 +706,39 @@ defmodule Castmill.Resources do
     with an existing entry.
   """
   def add_channel_entry(channel_id, entry_attrs \\ %{}) do
+    # Convert timestamps to DateTime structs
+    entry_attrs =
+      entry_attrs
+      |> Map.update("start", nil, &DateTime.from_unix!(&1, :millisecond))
+      |> Map.update("end", nil, &DateTime.from_unix!(&1, :millisecond))
+
     %ChannelEntry{
       channel_id: channel_id
     }
     |> ChannelEntry.changeset(entry_attrs)
     |> Repo.insert()
   end
 
+  @doc """
+    Get a channel entry.
+  """
+  def get_channel_entry(id) do
+    ChannelEntry
+    |> where(id: ^id)
+    |> Repo.one()
+  end
+
   @doc """
     Update entry in channel.
   """
   def update_channel_entry(%ChannelEntry{} = entry, attrs) do
```

Comment:
I guess we could end up with overlapping channel entries here as well. Maybe we need to wrap it in a transaction

Reply:
yes, but I am not sure how transactions can be used with change sets.